pkg/controllers/management/drivers/nodedriver/machine_driver.go

```diff
@@ -151,6 +219,10 @@ func (m *Lifecycle) Updated(obj *v3.NodeDriver) (runtime.Object, error) {
 		return obj, err
 	}
 
+	if err := m.createOrUpdateNodeForEmbeddedTypeCredential(obj.Spec.DisplayName+"credentialconfig", obj.Spec.DisplayName+"credentialConfig", obj.Spec.Active); err != nil {
```

Comment:
we should also extract name generation to its own method as it's used multiple times.